packages/salesforce-adapter/src/fetch.ts

```diff
@@ -128,16 +128,31 @@ const getInstanceFromMetadataInformation = (metadata: MetadataInfo,
 }
 
 export const fetchMetadataInstances = async ({
-  client, metadataType, fileProps, metadataQuery,
+  client, metadataType, fileProps, metadataQuery, maxInstancesPerType,
 }: {
   client: SalesforceClient
   fileProps: FileProperties[]
   metadataType: ObjectType
   metadataQuery: MetadataQuery
+  maxInstancesPerType?: number
 }): Promise<FetchElements<InstanceElement[]>> => {
   if (fileProps.length === 0) {
     return { elements: [], configChanges: [] }
   }
+
+  const typeName = fileProps[0].type
+  const instancesCount = fileProps.length
+  // We exclude metadataTypes with too many instances to avoid unwanted big and slow requests
+  if (typeName !== CUSTOM_OBJECT && maxInstancesPerType && instancesCount > maxInstancesPerType) {
```

Comment:
nit: we try to be more explicit when check if a value is defined
```suggestion
  if (typeName !== CUSTOM_OBJECT && maxInstancesPerType !== undefined && instancesCount > maxInstancesPerType) {
```